### ISSUE TITLE
Option to have zeros/ones/full output tensor with zero strides

### ISSUE THREAD

**Issue by kmbriedis** [pytorch/pytorch#38470]
## 🚀 Feature

An option to return a tensor with all strides set to 0 for `torch.zeros`, `torch.ones`, `torch.Tensor.new_ones`, `torch.Tensor.new_zeros`, etc. for smaller memory usage and better performance.

## Motivation

For now, an alternative way of doing this is by calling two methods, e.g.:
`x.new_full((1,), value).expand_as(x)`

But it is a pretty unfriendly approach, therefore, I have mostly seen a simple `torch.full_like(x, value)` being used instead, which can unnecessarily allocate a huge chunk of the memory.

## Pitch

Possible implementations:
* The most user-friendly way I can think of is to output zero-strided tensor by default and allocate additional memory only when a different value is assigned to some elements, but I can see this breaking many C++/CUDA extensions that expect a contiguous output from these methods (this could still be deprecated and changed in far future).
* Adding an additional [`torch.memory_format`](https://pytorch.org/docs/stable/tensor_attributes.html#torch.torch.memory_format) or  [`torch.layout`](https://pytorch.org/docs/stable/tensor_attributes.html#torch.torch.layout) that handles this
* Adding an additional argument for each of the functions

## Additional context

A real-life example where this would have been useful in the [NVIDIA SPADE code with a  dedicated method](https://github.com/NVlabs/SPADE/blob/0ff661e70131c9b85091d11a66e019c0f2062d4c/models/networks/loss.py#L51) for what could be potentially done as simple as `torch.zeros_like(input, expanded=False)`

**Comment by vadimkantorov** [comment]
`torch.full` would be nice as well. a real-world usage example: https://github.com/pytorch/pytorch/pull/36149/files#diff-c5db6ae353758a64f3e73dd758728cb7R139
